src/comment.ts

```diff
@@ -428,23 +488,30 @@ function categorize(
   head: Outcomes[string],
   base?: Outcomes[string],
 ): OutcomeCategory {
+  if (head.commit?.status !== "completed") {
+    return "pending";
+  }
+
   // Check for fatal failures first
-  switch (head.commit.completed?.conclusion) {
+  switch (head.commit.completed.conclusion) {
     case "fatal":
     case "timed_out":
       return "failure";
     case "merge_conflict":
     case "upstream_merge_conflict":
       return "merge_conflict";
-  }
-
-  // If not a completed success outcome, skip categorization
-  if (
-    !["error", "warning", "note", "success"].includes(
-      head.commit.completed?.conclusion || "",
-    )
-  ) {
-    return "pending";
+    case "noop":
+      return "success";
+    // Completed success outcomes are handled below
+    case "error":
+    case "warning":
+    case "note":
+    case "success": {
+      break;
+    }
+    // Unknown conclusions are fatal
+    default:
+      return "failure";
   }
 
   // Check for pending steps
```

Comment:
we should make it so that it's only pending, if everything is pending w/ no further notes.